dask_expr/io/parquet.py

```diff
@@ -714,54 +840,51 @@ def fragments(self):
             return list(ds.get_fragments(filter=pq.filters_to_expression(self.filters)))
         return self._dataset_info["fragments"]
 
-    @staticmethod
-    def _fragment_to_pandas(fragment, columns, filters, schema):
-        from dask.utils import parse_bytes
-
-        if isinstance(filters, list):
-            filters = pq.filters_to_expression(filters)
-        # TODO: There should be a way for users to define the type mapper
-        table = fragment.to_table(
-            schema=schema,
-            columns=columns,
-            filter=filters,
-            # Batch size determines how many rows are read at once and will
-            # cause the underlying array to be split into chunks of this size
-            # (max). We'd like to avoid fragmentation as much as possible and
-            # and to set this to something like inf but we have to set a finite,
-            # positive number.
-            # In the presence of row groups, the underlying array will still be
-            # chunked per rowgroup
-            batch_size=10_000_000,
-            # batch_readahead=16,
-            # fragment_readahead=4,
-            fragment_scan_options=pa.dataset.ParquetFragmentScanOptions(
-                pre_buffer=True,
-                cache_options=pa.CacheOptions(
-                    hole_size_limit=parse_bytes("4 MiB"),
-                    range_size_limit=parse_bytes("32.00 MiB"),
-                ),
-            ),
-            # TODO: Reconsider this. The OMP_NUM_THREAD variable makes it harmful to enable this
-            use_threads=True,
-        )
-        df = table.to_pandas(
-            types_mapper=_determine_type_mapper(),
-            use_threads=False,
-            self_destruct=True,
-        )
-        return df
-
     def _filtered_task(self, index: int):
         return (
-            ReadParquetPyarrowFS._fragment_to_pandas,
-            self.fragments[index],
+            _fragment_to_pandas,
+            FragmentWrapper(self.fragments[index]),
             self.columns,
             self.filters,
-            self._dataset_info["schema"],
+            self._dataset_info["schema"].remove_metadata(),
         )
 
 
+def _fragment_to_pandas(fragment_wrapper, columns, filters, schema):
```

Comment:
making this a proper function was not strictly necessary but I wanted to be extra sure that there is no weird reference. Also, I find this a little cleaner but it doesn't really matter, it's just style